src/lib/nvme_pt_ioctl.c

```diff
@@ -1580,23 +1580,22 @@ static int opal_read_datastr(int fd, struct opal_device *dev, uint8_t *data, uin
 	return ret;
 }
 
-static int list_lr(int fd, struct opal_device *dev)
+static int list_lr(int fd, struct opal_device *dev, struct sed_opal_lockingrange_arr *lrs)
```

Comment:
opal spec supports Global range plus at least 8 configurable ranges. This function will crash for a device having 11>= total ranges. This function at least should have protection that data will not be written outside of sed_opal_lockingrange_arr.